summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -204,49 +202,58 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
     }
     val metrics = getOrElse(stormDag, node, DEFAULT_SPOUT_STORM_METRICS)
 
-    val registerAllMetrics = Externalizer({ context: TopologyContext =>
+    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
+    val stormSpout = summerOpt match {
+      case Some(s) => {
+        val builder = BuildSummer(this, stormDag, node, jobID)
+        val registerAllMetricsAndCounters = createRegistrationFunction(metrics, getCountersForJob(jobID), jobID)
+        createSpoutToFeedSummer[Any, Any](stormDag, s, builder, tormentaSpout, registerAllMetricsAndCounters)
+      }
+      case None => {
+        val registerAllMetricsAndCounters = createRegistrationFunction(metrics, getCountersForJob(jobID), jobID)
+        getStormSpout(tormentaSpout, registerAllMetricsAndCounters)
+      }
+    }
+    topologyBuilder.setSpout(nodeName, stormSpout, sourceParallelism)
+  }
+
+  // Tormenta bug when openHook might get lost if it doesn't happen right before the getSpout.
+  private def getStormSpout[T](spout: Spout[T], register: Externalizer[(TopologyContext) => Unit]) = {
+    val hookedSpout = spout.openHook(register.get)
+    hookedSpout.getSpout
+  }
+
+  private def getCountersForJob(jobId: JobId) = JobCounters.getCountersForJob(jobId).getOrElse(Nil)
+
+  /*
+  This function takes the metrics and counters to register to the TopologyContext of a JobID and outputs a function which can be called on openHook.
+   */
+  private def createRegistrationFunction(metrics: SpoutStormMetrics, counters: TraversableOnce[(Group, Name)], jobID: JobId): Externalizer[(TopologyContext) => Unit] = {
```

Comment:
The `counters` we're passing is actually a `Seq`:

https://github.com/twitter/summingbird/blob/v0.11.0-RC1/summingbird-core/src/main/scala/com/twitter/summingbird/Stats.scala#L96

Lets make it a `Seq` here and avoid the `toSeq` call below.